### PR TITLE
WebXRManager: Fix for hand controllers

### PR DESCRIPTION
**Pull request by cabanier**
When the hand controllers go in and out of view, the browser is allowed to remove them from the input sources list.
The current implementation in the Oculus quest alway returns them, but that is not true going forward or in devices such as the Hololens.

On those devices, the right hand will be assigned the left controller which will cause it to deform.

This fix addresses that issues by looking up the correct controller instead of blindly using the index.

### CODE REVIEW

File: src/renderers/webxr/WebXRManager.js [hunk 1]
```diff
@@ -327,11 +327,12 @@ class WebXRManager extends EventDispatcher {
...
 			const inputSources = session.inputSources;
 
-			// Assign inputSources to available controllers
+			// Assign controllers to available inputSources
 
-			for ( let i = 0; i < controllers.length; i ++ ) {
+			for ( let i = 0; i < inputSources.length; i ++ ) {
 
-				inputSourcesMap.set( inputSources[ i ], controllers[ i ] );
+				const offset = inputSources[ i ].handedness === 'right' ? 1 : 0;
```
Comment (mrdoob) [inline src/renderers/webxr/WebXRManager.js:334]: `const index` may be more clear?

**Comment by mrdoob** [comment]
Thanks!

**Comment by hybridherbst** [comment]
@cabanier this sounds very similar to https://github.com/mrdoob/three.js/issues/23187 and the related PR https://github.com/mrdoob/three.js/pull/23188 - do you think it's the same issue? We also had to fix this for Hololens and also for phone-AR multitouch but it never got merged...

**Comment by cabanier** [comment]
@hybridherbst yes, this is the same issue and I verified that the same problem existed on the Hololens.
We're using the same backend as HL which is likely why both devices have this problem.

There's still an issue where the pointing ray gets duplicated every time a hand goes out of view. I didn't get a chance to look into that yet.

**Pull request merged**